scripts/src/javatests/org/oppia/android/scripts/coverage/CoverageReporterTest.kt

```diff
@@ -1,78 +1,291 @@
+      "${tempFolder.root}" +
+        "$coverageDir/CoverageReport.md"
+    ).readText()
+
+    val expectedMarkdown = buildString {
+      append("## Coverage Report\n\n")
+      append("- Number of files assessed: 1\n")
+      append("- Coverage Analysis: **PASS** :white_check_mark:\n\n")
+      append("<details>\n")
+      append("<summary>Succeeded Coverages</summary><br>\n\n")
+      append("| File | Coverage | Lines Hit | Status | Min Required |\n")
+      append("|------|:--------:|----------:|:------:|:------------:|\n")
+      append(
+        "| [$filename]($oppiaDevelopGitHubLink/$filename) " +
+          "| 100.00% | 10 / 10 | :white_check_mark: | $MIN_THRESHOLD% |\n"
+      )
+      append("</details>")
+    }
+
+    assertThat(outputReportText).isEqualTo(expectedMarkdown)
   }
 
   @Test
-  fun testCoverageReporter_validData_generatesCorrectCoverageRatio() {
-    reporter = CoverageReporter(
+  fun testGenerateMarkDownReport_withFailCoverageReportDetails_generateMarkdownTable() {
+    val oppiaDevelopGitHubLink = "https://github.com/oppia/oppia-android/tree/develop"
+    val coverageDir = "/coverage_reports"
+    val filename = "SampleFile.kt"
+    val validCoverageReport = CoverageReport.newBuilder()
+      .setDetails(
+        CoverageDetails.newBuilder()
+          .setFilePath(filename)
+          .setLinesFound(10)
+          .setLinesHit(0)
+          .build()
+      ).build()
+
+    val coverageReportContainer = CoverageReportContainer.newBuilder()
+      .addCoverageReport(validCoverageReport)
+      .build()
+
+    CoverageReporter(
       tempFolder.root.absolutePath,
-      validCoverageReport,
+      coverageReportContainer,
       ReportFormat.MARKDOWN
-    )
-    val expectedCoverageRatio = 0.8F
-    val (coverageRatio, _) = reporter.generateRichTextReport()
-    assertThat(expectedCoverageRatio).isEqualTo(coverageRatio)
+    ).generateRichTextReport()
+
+    val outputReportText = File(
+      "${tempFolder.root}" +
+        "$coverageDir/CoverageReport.md"
+    ).readText()
+
+    val expectedMarkdown = buildString {
+      append("## Coverage Report\n\n")
+      append("- Number of files assessed: 1\n")
+      append("- Coverage Analysis: **FAIL** :x:\n\n")
+      append("| File | Coverage | Lines Hit | Status | Min Required |\n")
+      append("|------|:--------:|----------:|:------:|:------------:|\n")
+      append(
+        "| [$filename]($oppiaDevelopGitHubLink/$filename) | " +
+          "0.00% | 0 / 10 | :x: | $MIN_THRESHOLD% |"
+      )
+    }
+
+    assertThat(outputReportText).isEqualTo(expectedMarkdown)
   }
 
   @Test
-  fun testCoverageReporter_noLinesFound_generatesZeroCoverageRatio() {
-    val expectedZeroCoverageRatio = 0F
-    // to check divided by zero error doesn't occur
-    val report = validCoverageReport.toBuilder().setLinesFound(0).build()
-    reporter = CoverageReporter(
+  fun testGenerateMarkDownReport_withFailureCoverageReportDetails_generateMarkdownTable() {
```

Comment:
```suggestion
  fun testGenerateMarkDownReport_withFailureCoverageReportDetails_generatesMarkdownTable() {
```

Note the verb tense for the "expected outcome" portion of the test name. This is the verb tense we should be using. Ditto for all other tests in all other test files in this PR.

Reply:
Ah Got it! Updated over here, will look out for any other test cases too.